Merge master into staging.

charlie-ht opened the pull request:
Please let me know what tests I should run, I tried running some tests with a debug build but that seems hopeless on both master and staging.

# Commits merged

[MSE] Fix BufUnbuf test from YouTube's MSE conformance test suite
06b35a84
    [MSE] Improve SourceBuffer::appendBuffer() time
e5b8d88b
    [MSE] Improve perfomance of SourceBuffer::evictCodedFrames
ce724e50
    [MSE] fix calculation of MediaSource::currentTimeFudgeFactor
26a5288a
    Cap the reported progress of implicit animations
7162d059
    [MSE] Fix unsigned comparison warning.
aec09ae1
    [GStreamer][MSE] Make setAppendState thread safe
4508db9f63a282e16e6f7b17902bcfb482be0d06
    [WPE] Enable MEMORY_SAMPLER
26614e8c4
  Revert "[GStreamer][MSE] Add missing lock around getStreamByTrackId"
aaab302388b1e75c8c085b184b1c0800557e3a6b
    Use the return value of gst_mini_object_make_writable...
b4df1c53648b6feca27351d0f00416fa8c080a34
    webbrige sets WPE_RAM_SIZE to 128M, which tickles a logic error in
    the cache model.
066c52ba8c8c7b9fd9ec0d10381765f9ef3de547
    Fixed UI issues in webinspector
b0e7fd855c5c81d307385664c8307677db1c46c9
    Fix for issue #326 WebInspector-Popover
9cbbdd7a287e6dcdb99fb1186b26c78711242a46
    [WPEWebkit] Added support for ARMV8 processor.
d3a336523d123119fe1dd53da5d9006c92cf078c
    [GStreamer] Fix video resize problems after quality change
b2ce7dc10c988996f956acbec2616b88e6fb1dd8
    Fix for PageUp/PageDown keys
29c10d9b0081d4941a6fce3ede49ad0fa36f6975
    [Gstreamer] Make sure live stream is detected
ef9a398ede7753ee5d7937f7304ce81c37c7f4c9
    Fix for WebInspector-NetworkTab
71d0d30c449b6b4d083ebc83e5880188010518db
    [GStreamer] Fix for invalid video size information sometimes returned by getVideoSizeAndFormatFromCaps (gst_video_info_from_caps)
59fbb5733a7b51ea98e35e157dd9c0a3bfe27b32
    [GStreamer] Make volume control consistent
939aa762744b9d84cd31a99edb5ad20a0a555c66
    [MSE][GStreamer] Fix premature video clip change on youtube
8dfa02e0439332db9f79771001fa832ddcd9da3f
    [GStreamer][MSE] Don't flush audio streams, only video streams
148ffdd91573a551308698b373b0f621943e0c4d
    Fixed bug with unneeded rangeRemoval
25c3dd3f60dbc03adbf16389fb91d37d15893687
    [MSE][GStreamer] don't pause playback when there is enough data
9842bc6629b797215502ceaa0911311720de7358
    [MSE][GStreamer] increase the playback buffer size
72a5ffc40ad7a4e21912bca936a9097edf893459
    Fixed UI issues in webinspector
3f777e3a36e36eef4ca188370b02317615ea7a10
    Fix for wpe PageClient overrides
50002033d48f60497034ec1af1e0acd2f49c3545
    Fixed MediaElementsEvents test
97c32bef21441f30d78c8faa32e7ef30f726096c
    [MSE][GStreamer] Fix infinite loop in evictCodedFrames() when duration is infinite
c11d0d412f4c06812f1823ac1c2115a4449ef57d
    Fixed typo indefiniteTime -> isIndefinite (#306)
a4cc7393e04476a6f33bd4569ef85b068a0fdaa8
    [MSE] [GStreamer] Fixing QUOTA_EXCEEDED_ERR fired when there was evictable data in Source Buffer (#296)
eec35dd7f6fffb78c556c519d5f208bd6c03813a
    minor compile errors [sic]
d392c150c13fbdff8e58e129ef345ff6dcb799e5
    volume method wrapped with WPE, usage is not
4e9b1cf8aa50bf6d26ce59581857be336fc2d9ab
    Add a METROLOGICAL define to be able to ifdef pieces of code specific to downstream WPE
fdac418f9f95f33ec29574d9872043099babc987
    Implemented custom headers per url
21f3896682f724cf7cd4b4e4dc8d4bf1e778ae7b
    Made fudge factor for audio a fraction
c527870398371a52a8df5673883f63b751c2d484
    Fix MSE video size test failures (#276) (#277)
277e5de9ba284592b98a6c6d28cbc7f7edcfd7dc
    [GStreamer][MSE] Reject too big channels/framerate/width/height mime type values

# Commits we might want but that conflict heavily or seem N/A.
254e68d1dcd1a9daa9861e6f12b1a7de6fe541bd
    Make scroll-to-focused-element behaviour configurable.
d608ad31fb4c66930e87ba40b4fadc24473ae97a
    Add <functional> header for std::functional
758f885ec6d52cf5564133646542b40c94a52543
    Fixed bug with wrong expires date in cookies
11727e2716b7a1be670643fc14fb245f28807483
    [MSE][GStreamer] keep the ahead buffer when evicting coded frames
7fb21a9b680482630b1cb8ee881666b34fb8d334
    [GStreamer][MSE] Fix stall in seek to end of buffered range
3cd4d31a10ecf2664388d622190652f1541613e4
    [GStreamer][MSE] Fix stall in seek due to incomplete reenqueue
b407c237 WKView: add a wpe_view_backend getter.
c99d5ee10f77dd02094d91d4645943e5c186c1bf
  [GStreamer] Notify player private (PlatformLayer) destruction to the
d9f7dc6f0edcbee70cdc7e640b83bff61ad30ccd
  Fix-for-delay-in-web-audio
06ff542aa8adbd1c27e8e8885afa344d104c12d1
    Generalise DB410c switch
427b0523d855395a5b0950d8a255383b9280c515
    [MSE] Don't try to adopt a floater.
259e898cb404ea0a9b90742ec1060aefec35190e
    Fix for 302 Redirect support (#356)
4e778b56897b69352ef81128d424c5ccd62b712f
    [WebInspector] Fix for elements page, use getPropertyValue instead of discontinued getPropertyCSSValue
313ff467f993bd46567b62276d10b0a0df33aa3a
    Add platform variable for DB410c
f017f70a2aad824eeb71e9fe19a2059065fdc6b8
    Using optimised vsink for platform DB410C
9a02007ed0010dc8adb5526d20aff7213f8324cb
    [GTK][WPE] border-radius with non visible border doesn't work on images that have their own RenderLayer
5674f3beca4d5a122b2793047660f709580795bd
    [GStreamer][WebAudio] Avoid sending null data/silence
83a08d0083c3db852313baa04ead30a6f2797ded
    Add a page property for enabling console log with private browsing
e2c897e6c35c0fd1c18bfa36c601f274765df723
    Use -ffunction-sections/-fdata-sections/--gc-sections with clang too [where has this stuff gone?]
cad2f3c7a0947cff44887232b800b40c4e30a79f
    Make Web Inspector Remote to listen on both IPv4 and IPv6 addresses
85504c3f6bde91f6697742982f72caa22a43c74d
    Make the code inclusive of musl when OS is linux
ae1b1e5d1e2a1e35a880a096823659ab4542a90a
    Use a setting to disable scrolling to the keyboard focused element instead of an env variable.

charlie-ht committed:
Merge master into staging.

Commits merged,

[MSE] Fix BufUnbuf test from YouTube's MSE conformance test suite
06b35a84
    [MSE] Improve SourceBuffer::appendBuffer() time
e5b8d88b
    [MSE] Improve perfomance of SourceBuffer::evictCodedFrames
ce724e50
    [MSE] fix calculation of MediaSource::currentTimeFudgeFactor
26a5288a
    Cap the reported progress of implicit animations
7162d059
    [MSE] Fix unsigned comparison warning.
aec09ae1
    [GStreamer][MSE] Make setAppendState thread safe
4508db9f63a282e16e6f7b17902bcfb482be0d06
    [WPE] Enable MEMORY_SAMPLER
26614e8c4
  Revert "[GStreamer][MSE] Add missing lock around getStreamByTrackId"
aaab302388b1e75c8c085b184b1c0800557e3a6b
    Use the return value of gst_mini_object_make_writable...
b4df1c53648b6feca27351d0f00416fa8c080a34
    webbrige sets WPE_RAM_SIZE to 128M, which tickles a logic error in
    the cache model.
066c52ba8c8c7b9fd9ec0d10381765f9ef3de547
    Fixed UI issues in webinspector
b0e7fd855c5c81d307385664c8307677db1c46c9
    Fix for issue #326 WebInspector-Popover
9cbbdd7a287e6dcdb99fb1186b26c78711242a46
    [WPEWebkit] Added support for ARMV8 processor.
d3a336523d123119fe1dd53da5d9006c92cf078c
    [GStreamer] Fix video resize problems after quality change
b2ce7dc10c988996f956acbec2616b88e6fb1dd8
    Fix for PageUp/PageDown keys
29c10d9b0081d4941a6fce3ede49ad0fa36f6975
    [Gstreamer] Make sure live stream is detected
ef9a398ede7753ee5d7937f7304ce81c37c7f4c9
    Fix for WebInspector-NetworkTab
71d0d30c449b6b4d083ebc83e5880188010518db
    [GStreamer] Fix for invalid video size information sometimes returned by getVideoSizeAndFormatFromCaps (gst_video_info_from_caps)
59fbb5733a7b51ea98e35e157dd9c0a3bfe27b32
    [GStreamer] Make volume control consistent
939aa762744b9d84cd31a99edb5ad20a0a555c66
    [MSE][GStreamer] Fix premature video clip change on youtube
8dfa02e0439332db9f79771001fa832ddcd9da3f
    [GStreamer][MSE] Don't flush audio streams, only video streams
148ffdd91573a551308698b373b0f621943e0c4d
    Fixed bug with unneeded rangeRemoval
25c3dd3f60dbc03adbf16389fb91d37d15893687
    [MSE][GStreamer] don't pause playback when there is enough data
9842bc6629b797215502ceaa0911311720de7358
    [MSE][GStreamer] increase the playback buffer size
72a5ffc40ad7a4e21912bca936a9097edf893459
    Fixed UI issues in webinspector
3f777e3a36e36eef4ca188370b02317615ea7a10
    Fix for wpe PageClient overrides
50002033d48f60497034ec1af1e0acd2f49c3545
    Fixed MediaElementsEvents test
97c32bef21441f30d78c8faa32e7ef30f726096c
    [MSE][GStreamer] Fix infinite loop in evictCodedFrames() when duration is infinite
c11d0d412f4c06812f1823ac1c2115a4449ef57d
    Fixed typo indefiniteTime -> isIndefinite (#306)
a4cc7393e04476a6f33bd4569ef85b068a0fdaa8
    [MSE] [GStreamer] Fixing QUOTA_EXCEEDED_ERR fired when there was evictable data in Source Buffer (#296)
eec35dd7f6fffb78c556c519d5f208bd6c03813a
    minor compile errors [sic]
d392c150c13fbdff8e58e129ef345ff6dcb799e5
    volume method wrapped with WPE, usage is not
4e9b1cf8aa50bf6d26ce59581857be336fc2d9ab
    Add a METROLOGICAL define to be able to ifdef pieces of code specific to downstream WPE
fdac418f9f95f33ec29574d9872043099babc987
    Implemented custom headers per url
21f3896682f724cf7cd4b4e4dc8d4bf1e778ae7b
    Made fudge factor for audio a fraction
c527870398371a52a8df5673883f63b751c2d484
    Fix MSE video size test failures (#276) (#277)
277e5de9ba284592b98a6c6d28cbc7f7edcfd7dc
    [GStreamer][MSE] Reject too big channels/framerate/width/height mime type values

Commits we might want but that conflict heavily or seem N/A.
254e68d1dcd1a9daa9861e6f12b1a7de6fe541bd
    Make scroll-to-focused-element behaviour configurable.
d608ad31fb4c66930e87ba40b4fadc24473ae97a
    Add <functional> header for std::functional
758f885ec6d52cf5564133646542b40c94a52543
    Fixed bug with wrong expires date in cookies
11727e2716b7a1be670643fc14fb245f28807483
    [MSE][GStreamer] keep the ahead buffer when evicting coded frames
7fb21a9b680482630b1cb8ee881666b34fb8d334
    [GStreamer][MSE] Fix stall in seek to end of buffered range
3cd4d31a10ecf2664388d622190652f1541613e4
    [GStreamer][MSE] Fix stall in seek due to incomplete reenqueue
b407c237 WKView: add a wpe_view_backend getter.
c99d5ee10f77dd02094d91d4645943e5c186c1bf
  [GStreamer] Notify player private (PlatformLayer) destruction to the
d9f7dc6f0edcbee70cdc7e640b83bff61ad30ccd
  Fix-for-delay-in-web-audio
06ff542aa8adbd1c27e8e8885afa344d104c12d1
    Generalise DB410c switch
427b0523d855395a5b0950d8a255383b9280c515
    [MSE] Don't try to adopt a floater.
259e898cb404ea0a9b90742ec1060aefec35190e
    Fix for 302 Redirect support (#356)
4e778b56897b69352ef81128d424c5ccd62b712f
    [WebInspector] Fix for elements page, use getPropertyValue instead of discontinued getPropertyCSSValue
313ff467f993bd46567b62276d10b0a0df33aa3a
    Add platform variable for DB410c
f017f70a2aad824eeb71e9fe19a2059065fdc6b8
    Using optimised vsink for platform DB410C
9a02007ed0010dc8adb5526d20aff7213f8324cb
    [GTK][WPE] border-radius with non visible border doesn't work on images that have their own RenderLayer
5674f3beca4d5a122b2793047660f709580795bd
    [GStreamer][WebAudio] Avoid sending null data/silence
83a08d0083c3db852313baa04ead30a6f2797ded
    Add a page property for enabling console log with private browsing
e2c897e6c35c0fd1c18bfa36c601f274765df723
    Use -ffunction-sections/-fdata-sections/--gc-sections with clang too [where has this stuff gone?]
cad2f3c7a0947cff44887232b800b40c4e30a79f
    Make Web Inspector Remote to listen on both IPv4 and IPv6 addresses
85504c3f6bde91f6697742982f72caa22a43c74d
    Make the code inclusive of musl when OS is linux
ae1b1e5d1e2a1e35a880a096823659ab4542a90a
    Use a setting to disable scrolling to the keyboard focused element instead of an env variable.

calvaris commented on Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp:
```diff
@@ -910,7 +926,15 @@ GstFlowReturn MediaPlayerPrivateGStreamerBase::newPrerollCallback(GstElement* si
+    if (info)
+        infoCopy = gst_structure_copy(info);
+    m_sample = adoptGRef(gst_sample_new(nullptr, gst_sample_get_caps(m_sample.get()),
+        gst_sample_get_segment(m_sample.get()), infoCopy));
```
info ? gst_structure_copy(info) : nullptr and we can avoid the infoCopy variable hussle

calvaris commented on Source/WebCore/platform/graphics/gstreamer/mse/AppendPipeline.cpp:
```diff
@@ -261,9 +260,27 @@ void AppendPipeline::handleNeedContextSyncMessage(GstMessage* message)
+                                                        "transition",
+                                                        G_TYPE_INT,
+                                                        AppendState::KeyNegotiation,
+                                                        nullptr);
```
This can be done in one line

calvaris commented on Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp:
```diff
@@ -318,19 +318,29 @@ bool MediaPlayerPrivateGStreamerMSE::doSeek()
+    if (!isTimeBuffered(seekTime) || m_readyState < MediaPlayer::HaveCurrentData) {
+        // Media source may trigger seek completion even when the target time is not yet buffered,
+        // in this case it is better continue the seek and wait for the app to provide media data.
+        m_mseSeekCompleted=true;
```
Need spaces around =

calvaris commented on Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp:
```diff
@@ -318,19 +318,29 @@ bool MediaPlayerPrivateGStreamerMSE::doSeek()
+            }
+            return true;
+        } else {
+            GST_DEBUG("[Seek] The target seek time is not buffered yet,"
```
Please, remove the brackets and put things in a single line.

calvaris commented on Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp:
```diff
@@ -902,19 +945,28 @@ void MediaPlayerPrivateGStreamerMSE::markEndOfStream(MediaSourcePrivate::EndOfSt
+    MediaTime playbackProgress = abs(position - cachedPosition);
 
-    if (m_eosPending && (paused() || (position >= durationMediaTime()))) {
+    if (m_eosPending && abs(position - durationMediaTime()) < MediaTime::createWithDouble(1) && !playbackProgress) {
```
Let's avoid createWithDouble here and create just (GST_SECOND/GST_SECOND)

calvaris commented on Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.h:
```diff
@@ -53,6 +53,8 @@ class MediaPlayerPrivateGStreamerMSE : public MediaPlayerPrivateGStreamer {
     void load(const String&) override;
     void load(const String&, MediaSourcePrivateClient*) override;
 
+    FloatSize naturalSize() const override;
```
Let's go final here.

calvaris commented on Source/WebKit/Shared/API/c/WKURLRequest.cpp:
```diff
@@ -70,3 +70,8 @@ void WKURLRequestSetDefaultTimeoutInterval(double timeoutInterval)
+void WKURLRequestSetHTTPHeaderField(WKURLRequestRef requestRef, WKStringRef key, WKStringRef value)
+{
+	toImpl(requestRef)->setHTTPHeaderField(toImpl(key)->string(), toImpl(value)->string());
+}
```
Maybe we can end the file in a new line.

calvaris commented on Source/WebCore/Modules/mediasource/SourceBuffer.cpp:
```diff
@@ -1730,6 +1803,21 @@ void SourceBuffer::sourceBufferPrivateDidReceiveSample(MediaSample& sample)
+
+    // To avoid playback pipeline starvation start providing media data as soon as we can
+    const AtomicString& trackID = sample.trackID();
+    auto it = m_trackBufferMap.find(trackID);
```
We probably want to auto& this.

charlie-ht replied:
The iterator or the trackId? Iterator doesn't make sense to me.

calvaris commented on Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp:
```diff
@@ -592,6 +599,13 @@ FloatSize MediaPlayerPrivateGStreamerBase::naturalSize() const
 
+    // Sanity check for the unlikely, but reproducible case when getVideoSizeAndFormatFromCaps returns incorrect values
+    if ((originalSize.width() == 0) || (originalSize.height() == 0)
+       || (pixelAspectRatioNumerator == 0) || (pixelAspectRatioNumerator == 0)) {
```
Numeric comparisons == 0 should be done as ! .

calvaris commented on Source/WebCore/platform/graphics/gstreamer/mse/AppendPipeline.cpp:
```diff
@@ -261,9 +260,27 @@ void AppendPipeline::handleNeedContextSyncMessage(GstMessage* message)
+    if (!g_strcmp0(contextType, "drm-preferred-decryption-system-id")) {
+        if (WTF::isMainThread()) {
+            transitionTo(AppendState::KeyNegotiation);
+        } else {
```
{ } not needed here.

calvaris commented on Source/WebCore/platform/graphics/gstreamer/mse/GStreamerMediaSample.h:
```diff
@@ -27,6 +27,7 @@
 #include "MediaSample.h"
 #include <gst/gst.h>
 #include <wtf/text/AtomicString.h>
+#include <wtf/PrintStream.h>
```
You need to sort these two includes.

calvaris replied:
Didn't the style checker complain about this?

calvaris commented on Source/WebCore/platform/graphics/gstreamer/WebKitWebSourceGStreamer.cpp:
```diff
@@ -622,7 +622,7 @@ static gboolean webKitWebSrcQueryWithParent(GstPad* pad, GstObject* parent, GstQ
 
             GstContext* context = gst_context_new("http-headers", FALSE);
-            gst_context_make_writable(context);
+            context = gst_context_make_writable(context);
```
:+1:

calvaris commented on Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp:
```diff
@@ -902,19 +945,28 @@ void MediaPlayerPrivateGStreamerMSE::markEndOfStream(MediaSourcePrivate::EndOfSt
+
 MediaTime MediaPlayerPrivateGStreamerMSE::currentMediaTime() const
 {
+    MediaTime cachedPosition = MediaTime::createWithFloat(m_cachedPosition);
```
no float here!

charlie-ht replied:
That would require refactoring the m_cachedPosition usage (my media time patch never made it into these branches, or upstream). I'm not going to refactor this while merging.

calvaris replied:
Ok, but please, do not lose track of that cause I think it is an important change.

calvaris commented on Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp:
```diff
@@ -902,19 +945,28 @@ void MediaPlayerPrivateGStreamerMSE::markEndOfStream(MediaSourcePrivate::EndOfSt
         m_isEndReached = true;
-        m_cachedPosition = m_mediaTimeDuration.toFloat();
+        position = m_mediaTimeDuration;
+        m_cachedPosition = position.toFloat();
```
no float here!

calvaris reviewed (changes requested):
We need to fix these things. staging is close to upstream meaning we want to respect style even more. I don't think you need to to fix these things in every commit where they happened, just a review commit at the end.

Also, if the changes I am talking about are needed upstream we should file bugs upstream.

Besides, I guess @eocanha, @zdobersek and @carlosgcampos should have a quick look at this too.

charlie-ht committed:
Fix style comments.

These are from calvaris and the check-webkit-style, ignoring all
warnings less than category level 5, because there's too many to fix
(and calvaris didn't notice most of them :))

calvaris commented on Source/WebCore/Modules/mediasource/SourceBuffer.cpp:
```diff
@@ -1810,11 +1807,10 @@ void SourceBuffer::sourceBufferPrivateDidReceiveSample(MediaSample& sample)
             trackBuffer.lastDecodeTimestamp.isValid() &&
-            abs(trackBuffer.lastEnqueuedDecodeEndTime - trackBuffer.lastDecodeTimestamp) > MediaTime::createWithDouble(0.350) )
-        {
+            abs(trackBuffer.lastEnqueuedDecodeEndTime - trackBuffer.lastDecodeTimestamp) > MediaTime::createWithDouble(0.350)) {
```
It seems that for this, { } are not needed.

calvaris commented on Source/WebCore/platform/graphics/gstreamer/mse/PlaybackPipeline.cpp:
```diff
@@ -113,7 +113,7 @@ MediaSourcePrivate::AddStatus PlaybackPipeline::addSourceBuffer(RefPtr<SourceBuf
     GST_DEBUG_OBJECT(m_webKitMediaSrc.get(), "State %d", int(GST_STATE(m_webKitMediaSrc.get())));
 
-    Stream* stream = new Stream{ };
+    Stream *stream = new Stream { };
```
Oops!

calvaris commented on Source/WebCore/Modules/mediasource/SourceBuffer.cpp:
```diff
@@ -868,14 +895,21 @@ void SourceBuffer::evictCodedFrames(size_t newDataSize)
+        return;
+    }
+
+    auto buffered = m_buffered->ranges();
```
Better auto&

calvaris commented on Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp:
```diff
@@ -592,6 +599,13 @@ FloatSize MediaPlayerPrivateGStreamerBase::naturalSize() const
+    if (!originalSize.width() || !originalSize.height()
+       || !pixelAspectRatioNumerator || !pixelAspectRatioNumerator) {
+        GST_DEBUG("getVideoSizeAndFormatFromCaps returned an invalid info, returning an empty size");
+        return FloatSize();
```
We can return { }.

calvaris commented on Source/WebCore/Modules/mediasource/MediaSource.cpp:
```diff
@@ -500,15 +500,18 @@ ExceptionOr<void> MediaSource::setDurationInternal(const MediaTime& duration)
-        for (auto& sourceBuffer : *m_sourceBuffers)
-            sourceBuffer->rangeRemoval(newDuration, m_duration);
+        for (auto& sourceBuffer : *m_sourceBuffers) {
+            auto length = sourceBuffer->bufferedInternal().length();
```
I think the auto here is extreme lazyness. We can go for unsigned.

calvaris commented on Source/WebCore/Modules/mediasource/MediaSource.cpp:
```diff
@@ -855,12 +858,32 @@ bool MediaSource::isTypeSupported(const String& type)
     MediaEngineSupportParameters parameters;
     parameters.type = contentType;
+    parameters.channels = channels;
+    parameters.dimension = FloatSize(width, height);
```
You can probably use the { } constructor here.

calvaris reviewed (changes requested):
Very nice style catches. We are much closer!

charlie-ht committed:
Fix more style issues.

The following is now silent,

./Tools/Scripts/check-webkit-style -g 90bc78479269d6ec283198aad79ef0161781d858..

charlie-ht commented:
I rebased over @zdobersek merge today and fixed the conflicts, also hope I've addressed the style comments now.

calvaris reviewed (approved):
If this does not break anything, we're good to go.

charlie-ht commented:
Thanks for the reviews calvaris.
